shumai/tensor/tensor.ts

```diff
@@ -416,7 +416,7 @@ export class Tensor {
     }
   }
 
-  checkpoint(file?: () => any, callback?: () => boolean) {
+  checkpoint(file?: () => any | any, callback?: () => boolean) {
```

Comment:
mistake or is this doing some voodoo I'm not aware of?

Reply:
Ahh, good catch! Had been toying with trying to name all possible types, but yea that's an error. I'll fix that now!

Reply:
Fixed; changed the conditional to `if (file instanceof Function)` as that should be marginally more performant.